spec/models/service_template_spec.rb

```diff
@@ -50,20 +50,24 @@
       false_expression = MiqExpression.new("=" => {"field" => "ServiceTemplate-name", "value" => "bar"})
       FactoryGirl.create(:custom_button,
                          :name                  => "visible button",
-                         :applies_to_class      => "Service",
+                         :applies_to_class      => "ServiceTemplate",
+                         :applies_to_id         => service_template.id,
```

Comment:
This shouldn't be here - it (and the ones below) are intended to serve as examples of generic buttons, i.e. not associated to any record in particular

Reply:
I think we can not create generic button for `ServcieTemplate` using UI,
Also, `ServiceTemplate#generic_custom_buttons` returns generic buttons defined on `Service` :
https://github.com/ManageIQ/manageiq/blob/600392e6594fe55078ca9621f649de0806ef4966/app/models/service_template.rb#L485 To change test and use generic buttons here, - they should be defined on `Service` and `MiqExpression` should be modifyied to use `Service`model. 
However, there is already test with generic button on `Service`: https://github.com/yrudman/manageiq/blob/b6660c1d0279ab401861c156831ff29ad6f6a2c4/spec/models/service_template_spec.rb#L153 
and i wanted to show both examples: generic and direct buttons.

Reply:
@imtayadeway i have change this test to use generic buttons